app/controllers/result_filters/location_controller.rb

```diff
@@ -2,21 +2,89 @@ module ResultFilters
   class LocationController < ApplicationController
     include FilterParameters
 
+    NO_OPTION = nil
+    LOCATION_OPTION = "1".freeze
+
     def new; end
 
     def create
-      if(!params[:l])
-        flash[:error] = "Please choose an option"
-        redirect_to location_path(params_without_unsupported_location_params)
+      build_params = strip(filter_params.clone)
+
+      case selected_option
+      when NO_OPTION
+        handle_no_option(build_params)
+      when LOCATION_OPTION
+        if location_query.empty?
+          handle_missing_location(build_params)
+        else
+          handle_location_option(build_params)
+        end
       else
-        redirect_to results_path(params_without_unsupported_location_params)
+        redirect_to results_path(build_params)
       end
     end
 
   private
 
-    def params_without_unsupported_location_params
-      filter_params.except(:lat, :lng, :rad, :query, :loc, :lq)
+    def handle_location_option(build_params)
+      geocode_params = geocode_params_for(location_query)
+      if geocode_params
+        merged_params = build_params.merge(geocode_params)
+        redirect_to results_path(merged_params)
+      else
+        build_errors_for(:unknown_location)
+        redirect_to location_path(build_params)
+      end
+    end
+
+    def handle_missing_location(build_params)
+      build_errors_for(:missing_location)
+      redirect_to location_path(build_params)
+    end
+
+    def handle_no_option(build_params)
+      build_errors_for(:none_selected)
+      redirect_to location_path(build_params)
+    end
+
+    def build_errors_for(type)
+      flash[:query] = location_query
+
+      case type
+      when :none_selected
+        flash[:error] = "Please choose an option"
+      when :missing_location
+        flash[:error] = "Postcode, town or city"
+        flash[:location_error] = "Please enter a postcode, city or town in England"
+      when :unknown_location
+        flash[:error] = "Postcode, town or city"
+        flash[:location_error] = "We couldn't find this location, please check your input and try again."
+      end
+    end
+
+    def strip(params)
+      params.reject { |_, v| v == "" }
+    end
+
+    def geocode_params_for(query)
+      results = Geocoder.search(query, components: "country:UK")
+      england_results = results.select { |r| r.state == "England" }.first
```

Comment:
https://github.com/DFE-Digital/search-and-compare-api/blob/3aa6be82dcf5548892580e900ea1e37decf63a0f/src/api/Services/Geocoder/TECH_DEBT__TemporarilyCopied__Geocoder.cs#L61-L65

Not too sure about tying it down to `England` only

Reply:
But the error messages / guidance say to enter a `postcode in, city or town in England`

Reply:
New bug ticket has been created for error messages